bootstrap/bootstrap-storage.sh

```diff
@@ -49,41 +49,42 @@ done
 
 echo "Using Configuration file: ${CONFIG_FILE_PATH}"
 
-function dropTables {
-    ${JAVA} -Dbootstrap.dir=$BOOTSTRAP_DIR  -cp ${CLASSPATH} ${TABLE_INITIALIZER_MAIN_CLASS} -m ${MYSQL_JAR_URL_PATH} -c ${CONFIG_FILE_PATH} -s ${SCRIPT_ROOT_DIR} --drop
-}
-
-function createTables {
-    ${JAVA} -Dbootstrap.dir=$BOOTSTRAP_DIR  -cp ${CLASSPATH} ${TABLE_INITIALIZER_MAIN_CLASS} -m ${MYSQL_JAR_URL_PATH} -c ${CONFIG_FILE_PATH} -s ${SCRIPT_ROOT_DIR} --create
-}
-
-function checkStorageConnection {
-    ${JAVA} -Dbootstrap.dir=$BOOTSTRAP_DIR  -cp ${CLASSPATH} ${TABLE_INITIALIZER_MAIN_CLASS} -m ${MYSQL_JAR_URL_PATH} -c ${CONFIG_FILE_PATH} -s ${SCRIPT_ROOT_DIR} --check-connection
+function execute {
+    ${JAVA} -Dbootstrap.dir=$BOOTSTRAP_DIR  -cp ${CLASSPATH} ${TABLE_INITIALIZER_MAIN_CLASS} -m ${MYSQL_JAR_URL_PATH} -c ${CONFIG_FILE_PATH} -s ${SCRIPT_ROOT_DIR} --${1}
 }
 
 function printUsage {
-    echo "USAGE: $0 [create|drop|check-connection|drop-create]"
+    cat <<-EOF
+    USAGE: $0 [create|migrate|info|validate|drop|repair|check-connection]
+       create   : Applies all the migrations from scratch, the target database has to be empty for this option
+       migrate  : Applies all the migration on the target database from the last check point, the target database should not be empty for this option
+       info     : Shows the list of migrations applied and the pending migration waiting to be applied on the target database
+       validate : Checks if the all the migrations haven been applied on the target database
+       drop     : Drops all the tables in the target database
+       repair   : Repairs the DATABASE_CHANGE_LOG table which is used to track all the migrations on the target database.
+                  This involves removing entries for the failed migrations and update the checksum of migrations already applied on the target databsase.
+       check-connection : checks if a connection can be sucessfully obtained for the target database
+EOF
 }
 
-opt="create"
-
-if [ $# -gt 0 ]
+if [ $# -gt 1 ]
 then
-    opt="$1"
+    echo "More than one argument specified, please use only one of the below options"
+    printUsage
+    exit 1
 fi
 
+opt="$1"
+
 case "${opt}" in
-create)
-    createTables
-    ;;
-drop)
-    dropTables
+create | drop | migrate | info | validate | repair )
+    execute ${opt}
     ;;
-drop-create)
-    dropTables && createTables
+drop-create )
+    execute drop && execute create
     ;;
-check-connection)
-    checkStorageConnection
+check-connection )
```

Comment:
Depending upon the exist status of the execute command, we print a meaningful message to the console. We can make the exception throw by the Tableinitializer class more readable and move this option to the first case condition.